sonar-php-plugin/src/main/java/org/sonar/plugins/php/PHPRulesDefinition.java

```diff
@@ -32,7 +42,77 @@
   public void define(Context context) {
     NewRepository repository = context.createRepository(CheckList.REPOSITORY_KEY, Php.KEY).setName(REPOSITORY_NAME);
     new AnnotationBasedRulesDefinition(repository, Php.KEY).addRuleClasses(false, CheckList.getChecks());
+
+    for (NewRule rule : repository.rules()) {
+      String metadataKey = rule.key();
+      // Setting internal key is essential for rule templates (see SONAR-6162), and it is not done by AnnotationBasedRulesDefinition from sslr-squid-bridge version 2.5.1:
+      rule.setInternalKey(metadataKey);
+      rule.setHtmlDescription(readRuleDefinitionResource(metadataKey + ".html"));
+      addMetadata(rule, metadataKey);
+    }
+
     repository.done();
   }
 
+  @Nullable
+  private static String readRuleDefinitionResource(String fileName) {
+    URL resource = PHPRulesDefinition.class.getResource("/org/sonar/l10n/php/rules/php/" + fileName);
+    if (resource == null) {
+      return null;
+    }
+    try {
+      return Resources.toString(resource, Charsets.UTF_8);
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to read: " + resource, e);
+    }
+  }
+
+  private void addMetadata(NewRule rule, String metadataKey) {
+    String json = readRuleDefinitionResource(metadataKey + ".json");
+    if (json != null) {
+      Gson gson = new Gson();
+      RuleMetadata metadata = gson.fromJson(json, RuleMetadata.class);
+      rule.setSeverity(metadata.defaultSeverity.toUpperCase(Locale.US));
+      rule.setName(metadata.title);
+      rule.setTags(metadata.tags);
+      rule.setType(RuleType.valueOf(metadata.type));
+      rule.setStatus(RuleStatus.valueOf(metadata.status.toUpperCase(Locale.US)));
+
+      if (metadata.remediation != null) {
+        // metadata.remediation is null for template rules
+        rule.setDebtRemediationFunction(metadata.remediation.remediationFunction(rule.debtRemediationFunctions()));
+        rule.setGapDescription(metadata.remediation.linearDesc);
+      }
+    }
+  }
+
+  private static class RuleMetadata {
+    String title;
+    String status;
+    String type;
+    @Nullable
+    Remediation remediation;
+
+    String[] tags;
+    String defaultSeverity;
+  }
+
+  private static class Remediation {
+    String func;
+    String constantCost;
+    String linearDesc;
+    String linearOffset;
+    String linearFactor;
+
+    private DebtRemediationFunction remediationFunction(DebtRemediationFunctions drf) {
+      if (func.startsWith("Constant")) {
+        return drf.constantPerIssue(constantCost.replace("mn", "min"));
```

Comment:
@ivandalbosco I see that you just copied this piece from JS, but it's not perfect there :) We can drop `.replace("mn", "min")` everywhere. This was done as many RSPECs had legacy `mn` for minutes. It should be fixed now. It there is still some RSPEC using `mn` and plugin fails, then you better update this RSPEC